portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,73 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.List;
+
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<Comment>comments = new ArrayList<>();
+    int userChoice = getUserChoice(request);
+    int count = 0;
+    for (Entity commentEntity : results.asIterable()) {
+      long tempId = (long) commentEntity.getKey().getId();
+      String tempComment = (String) commentEntity.getProperty("comment");
+      long tempTimestamp = (long) commentEntity.getProperty("timestamp");
+      if(tempComment != null && tempComment.strip()!= "" && count<userChoice){
+        Comment comment = new Comment(tempId, tempComment, tempTimestamp);
+        comments.add(comment);
+        count++;
```

Comment:
nit: Consider `break`-ing the for loop early if `count > userChoice`. Totally optional!

Reply:
Addressed in latest commit. Now no "if" statement has more than two boolean conditions.